database/rrdset.c

```diff
@@ -519,19 +519,114 @@ RRDSET *rrdset_create_custom(
+                    freez(st->plugin_name);
+                    st->plugin_name = strdupz(plugin);
+                    mark_rebuild = 1;
+                }
+            } else {
+                if (plugin != st->plugin_name) { // one is NULL?
+                    if (st->plugin_name)
+                        freez(st->plugin_name);
+                    st->plugin_name = plugin ? strdupz(plugin) : NULL;
+                    mark_rebuild = 1;
+                }
+            }
+
+            if (module && st->module_name) {
+                if (unlikely(strcmp(module, st->module_name))) {
+                    freez(st->module_name);
+                    st->module_name = strdupz(module);
+                    mark_rebuild = 1;
+                }
+            } else {
+                if (module != st->module_name) {
+                    if (st->module_name)
+                        freez(st->module_name);
+                    st->module_name = module ? strdupz(module) : NULL;
+                    mark_rebuild = 1;
+                }
+            }
+
+            char *new_title = title ? strdupz(title) : NULL;
+            if (new_title) {
+                json_fix_string(new_title);
+                if (unlikely(strcmp(st->title, new_title))) {
+                    freez(st->title);
+                    st->title = new_title;
+                    mark_rebuild = 1;
+                } else
+                    freez(new_title);
+            }
+
+            RRDSET_TYPE new_chart_type =
+                rrdset_type_id(config_get(st->config_section, "chart type", rrdset_type_name(chart_type)));
+            if (st->chart_type != new_chart_type) {
+                st->chart_type = new_chart_type;
+                mark_rebuild = 1;
+            }
+
+            char *new_family = family ? strdup(family): NULL;
+            if (new_family) {
+                json_fix_string(new_family);
+                if (unlikely((strcmp(st->family, new_family)))) {
+                    freez(st->family);
+                    st->family = new_family;
+                    mark_rebuild = 1;
+                } else
+                    freez(new_family);
+            }
+
+            char *new_context = context ? strdup(context) : NULL;
+            if (new_context) {
+                json_fix_string(new_context);
+                if (unlikely((strcmp(st->context, new_context)))) {
+                    freez(st->context);
+                    st->context = new_context;
+                    st->hash_context = simple_hash(st->context);
+                    mark_rebuild = 1;
+                } else
+                    freez(new_context);
+            }
+
+            rrdhost_unlock(host);
+
+#ifdef ENABLE_DBENGINE
+            if (is_archived == 0 && st->rrd_memory_mode == RRD_MEMORY_MODE_DBENGINE && mark_rebuild) {
+                debug(D_METADATALOG, "CHART [%s] metadata updated", st->id);
+                metalog_commit_update_chart(st);
+            }
+#endif
```

Comment:
Should we do ACLK updates as well?

Reply:
There are a few checks to be done for this to work properly. Since a plugin / module can also be updated, that may delete and add a new collector. 

ACLK keeps track of the active collectors by counting their chart and  decides if a chart update should go out or a full `on-connect` payload.

Reply:
ACLK updates should not go out during metadata log replay

Reply:
`if (is_archived == 0)` means it's not during log replay though. Replay sets `is_archived = 1`.

Reply:
I made a couple of changes to handle switch from archived to active (which should also update the collector list within ACLK)